dali/python/nvidia/dali/external_source.py

```diff
@@ -419,12 +526,14 @@ def __call__(self, *, source = None, cycle = None, name = None, layout = None, c
             name = self._name
 
         if name is not None and self._num_outputs is not None:
-            raise RuntimeError("``num_outputs`` is not compatible with named ``ExternalSource``")
+            raise RuntimeError("``num_outputs`` is not compatible with named ``ExternalSource``.")
 
         if self._num_outputs is not None:
             outputs = []
-            kwargs = {}
-            group = _ExternalSourceGroup(callback, True, cuda_stream=cuda_stream, use_copy_kernel=use_copy_kernel, batch=batch)
+            kwargs = {"no_copy": no_copy}
+            group = _ExternalSourceGroup(
+                callback, True, cuda_stream=cuda_stream, use_copy_kernel=use_copy_kernel,
+                batch=batch, parallel=parallel, prefetch_queue_depth=prefetch_queue_depth)
```

Comment:
How about putting common arguments to this call and the one in `else` branch into a dict and unpacking it here with `**`? This will save us from potential bugs when we forget to update one of the branches.

Reply:
Done